File: soal6.js
// Data email peserta
const emails = [
  "Andi@gmail.com",
  "Budi@gmail.com",
  "Sari@gmail.com",
  "Andi@gmail.com",
  "Rina@gmail.com",
  "Budi@gmail.com"
];

// Object untuk menghitung jumlah kemunculan
let count = {};
let duplicates = [];

// Loop untuk menghitung
for (let i = 0; i < emails.length; i++) {
  let email = emails[i];

  if (count[email]) {
    count[email]++;
  } else {
    count[email] = 1;
  }
}

// Cek mana yang lebih dari 1
for (let email in count) {
  if (count[email] > 1) {
    duplicates.push(email);
  }
}

// Output
if (duplicates.length > 0) {
  console.log("Email duplikat ditemukan:");
  duplicates.forEach(email => console.log("- " + email));
} else {
  console.log("Tidak ada data duplikat.");
}

File: soal6.py
# Menentukan jumlah email yang akan diinput
jumlah_email = int(input("Masukkan jumlah email yang ingin dicek: "))

emails = []

# Input email
for i in range(jumlah_email):
    email = input(f"Masukkan email ke-{i+1}: ")
    emails.append(email)

# Set untuk menyimpan email unik dan duplikat
email_unik = set()
duplikat = set()

# Proses pengecekan duplikat
for email in emails:
    if email in email_unik:
        duplikat.add(email)
    else:
        email_unik.add(email)

# Menampilkan hasil
print("\n=== Hasil Pengecekan Duplikat ===")
if duplikat:
    print("Email yang terdeteksi duplikat:")
    for email in duplikat:
        print("-", email)
else:
    print("Tidak ada data duplikat.")
